astropy/io/ascii/tests/test_read.py

```diff
@@ -158,6 +161,50 @@ def test_read_all_files_via_table(fast_reader):
                 assert_equal(len(table[colname]), testfile['nrows'])
 
 
+@pytest.mark.parametrize('fast_reader', [True, False, 'force'])
+def test_read_all_files_with_encoding(fast_reader, tmpdir):
+    from ....table import Column
+    formats = {v: k for k, v in core.FORMAT_CLASSES.items()}
+
+    for testfile in get_testfiles():
+        if testfile.get('skip'):
+            print('\n\n******** SKIPPING %s' % testfile['name'])
+            continue
+
+        tmpfile = str(tmpdir.join(os.path.basename(testfile['name'])))
```

Comment:
Add a comment here giving guidance on the overall plan of generating a new table with an encoded column where possible.